operator/internal/manifests/env_vars_test.go

```diff
@@ -0,0 +1,97 @@
+package manifests
+
+import (
+	"strings"
+	"testing"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestContainerEnvVars_ReadVarsFromCustomResource(t *testing.T) {
```

Comment:
Should we add a test for "the container already has one of the environment variables but we are overriding it"?